### DISCUSSION TITLE
Running SExtractor

### DISCUSSION THREAD

**Discussion by Astrogirlanajli** [astromatic/sextractor#73] [Q&A]
Hi, After installing the SExtractor, I tried running SExtractor for test image(galaxies.fits), it's giving me the error:
(mirar) anjali@RSAA-024587 ~ % sex galaxies.fits
----- SExtractor 2.28.0 started on 2024-09-20 at 00:48:16 with 1 thread

> Setting catalog parameters
> *ERROR*: can't read default.param
I installed SExtractor using github clone method:
git clone https://github.com/astromatic/sextractor.git
cd sextractor
./autogen.sh
./configure --disable-model-fitting
make
sudo make install
I wanted to know how to fix the error and run SExtractor.

**Comment by kirxkirx** [comment]
SExtractor needs a few configuration files to run. These configuration files are often located in the same directory where one runs SExtractor. Typically one copies these configuration files from a previous SExtractor run and modifies them as needed, but here is an example of how one could make SExtractor create a set of configuration files from scratch:
````
# Write default SExtractor parameter file
sex -d > default.sex
# Create default image convolution file
echo "CONV NORM
# 3x3 ``all-ground'' convolution mask with FWHM = 2 pixels.
1 2 1
2 4 2
1 2 1" > default.conv
# Write SExtractor output source parameters file uncommentating some output parameters on the fly
sex -dp | sed 's/#NUMBER/NUMBER/' | sed 's/#MAG_AUTO/MAGERR_AUTO/' | sed 's/#MAG_AUTO/MAGERR_AUTO/' | sed 's/#X_IMAGE/X_IMAGE/' | sed 's/#Y_IMAGE/Y_IMAGE/' > default.param
# at this point SExtractor is ready for a test run
####################################################
# Run SExtractor on my test FITS file f_72-001r.fit located in the directory current_work/sample_data/ within my home directory (~)
sex ~/current_work/sample_data/f_72-001r.fit
# Print out the list of detected sources in the terminal
cat test.cat
````
For a real science run one would definitely need to tweak the source detection parameters in `default.sex` for a given telescope/camera combination as well as the list of output parameters in `default.param`. Surprisingly, the default image convolution kernel `default.conv` works fine for a vast majority of ground-based telescopic images (source detection on images taken with a telephoto lens typically benefit from a narrower convolution kernel). The details about the source detection and output parameters may be found in [SExtractor documentation.](https://sextractor.readthedocs.io/en/latest/Introduction.html).

**Reply by Astrogirlanajli** [reply]
Although I created a .conv file with all ground kernels:


CONV NORM # 3x3 ``all-ground'' convolution mask with FWHM = 2 pixels.
1 2 1
2 4 2
1 2 1
After running sex -d > file. conv, and calling interest of target:
(base) anjali@RSAA-024587 extractor % sex -d > /Users/Anjali/extractor/config/gn.conv
(base) anjali@RSAA-024587 extractor % sex /Users/Anjali/github/Mirar/Sources/NGC.fits

I'm getting the following error:
----- SExtractor 2.28.0 started on 2024-09-27 at 11:08:09 with 1 thread

> Reading detection filter
> *Error*: not a suitable filter in default.conv

After executing the file, my file.conv is changing itself into this https://github.com/Astrogirlanajli/Research_thesis/blob/main/gn.conv file. I wonder why it's transforming itself. To be noted, I installed SExtractor via the GitHub source package, which is technically cloned github.

**Comment by Astrogirlanajli** [comment]
Hi, It's still showing an error as I run the SEx tractor command.


________________________________

**Reply by kirxkirx** [reply]
Is this the same error as before, "ERROR: can't read default.param"? If yes, please make sure that the file named `default.param` actually exists in the directory where you run SExtractor.You can check that the file is there by doing `ls default.param` in that directory. If the file is there, try to open it with a text editor or just print the content of the file in the terminal by running `cat default.param`. Make sure there are some parameters in default.param that are uncommented (do not start with `#`) so SExtractor has something to output.

If an error message is not about default.param but is different, please copy the error message here with a few lines above and below the error message. Maybe it will clarify what's going wrong...

**Comment by ebertin** [comment]
Hi @Astrogirlanajli,

please note that after doing a `sudo make install` on Linux systems, default configuration files, including `default.param` and `default.conv` can be found in `/usr/local/share/sextractor`.

**Reply by Astrogirlanajli** [reply]
Hi, I didn't install it via sudo make install; I had many issues while installing it on my Mac, so I cloned the SExtractor github.

**Comment by rmathar** [comment]
The main configuration file and the convolution kernel are two different pairs of shoes. The `sex -d > /Users/Anjali/extractor/config/gn.conv` creates a configuration (!) file and overwrites the previous content by the standard syntax of Linux bash commands. So if you generate `file.conv` and then type `sex -d > file.conv`, the `file.conv` is replaced and overwritten with a default configuration file; this is expected.

The standard way of getting consistent operation is (i) generate a configuration file with `sex -d > /Users/Anjali/extractor/config/gn.conf`, (ii) edit the configuration file `/Users/Anjali/extractor/config/gn.conf` so its line with `FILTER_NAME` has parameter `text.conv`, (iii) generate the convolution kernel file `text.conv` with your favorite ASCII editor, (iv) run `sex -c /Users/Anjali/extractor/config/gn.conf /Users/Anjali/github/Mirar/Sources/NGC.fits` so it looks inside the configuration file `/Users/Anjali/extractor/config/gn.conf` what the name of the convolution file is.
